### PR TITLE
fix: use a umoci which has fix for another empty lgetxattr

### PR DESCRIPTION
**Pull request by hallyn**
We end up with errors like:

/home/shallyn/stackertest2/roots/test2/overlay/etc/systemd/system/multi-user.target.wants: dropped overlay attrs: user.overlay.origin error: no data available
get xattr: user.overlay.origin
github.com/opencontainers/umoci/oci/layer.(*tarGenerator).AddFile
        /stacker-tree/.build/gopath/pkg/mod/github.com/project-stacker/umoci@v0.0.0-20231025233220-c7bb24d57e98/oci/layer/tar_generate.go:225
github.com/opencontainers/umoci/oci/layer.GenerateInsertLayer.func1.2

Unless there is an actual data i/o error, ignore it.  In particular, if the fs either says eopnotsupp, enodata, or eperm, just assume the xattr did not actually exist.  We have to do this because there are cases where listxattr will return xattr keys which getxattr will filter out, and we're expected to ignore those.

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hallyn** [comment]
I'm using my tree so builds can happen.  Obviously we'll want to get that fix back into github.com/project-stacker/umoci

File: go.sum [hunk 1]
```diff
@@ -508,6 +508,8 @@ github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1/go.mod h1:wJfORR
 github.com/gorilla/mux v1.8.1 h1:TuBL49tXwgrFYWhqrNgrUNEY92u81SPhu7sTdzQEiWY=
 github.com/gorilla/mux v1.8.1/go.mod h1:AKf9I4AEqPTmMytcMc0KkNouC66V3BtZ4qD5fmWSiMQ=
 github.com/grpc-ecosystem/grpc-gateway v1.16.0/go.mod h1:BDjrQk3hbvj6Nolgz8mAMFbcEtjT1g+wF4CSlocrBnw=
+github.com/hallyn/umoci v0.0.0-20240122192320-3c661c1beb44 h1:4xaZ4I8PCnycSDr0RHn0JJVJtuJr7Q852ui8ChOI8rQ=
```
Comment (andaaron) [inline go.sum:511]: Run a go mod tidy. Should update these lines and fix the compilation errors.

Comment (hallyn) [reply go.sum:511]: Will do, but note that I had done a go mod tidy, a full build, and a full test locally yesterday.  Seems fragile...

Comment (hallyn) [reply go.sum]: Done, hoping for the best, thanks :)

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**